src/python/dxpy/scripts/dx.py

```diff
@@ -3149,24 +3149,61 @@ def run_batch_all_steps(args, executable, dest_proj, dest_path, input_json, run_
 def run_body(args, executable, dest_proj, dest_path, preset_inputs=None, input_name_prefix=None):
     input_json = _get_input_for_run(args, executable, preset_inputs)
 
-    if args.sys_reqs_from_clone and not isinstance(args.instance_type, str):
-        args.instance_type = dict({stage: reqs['instanceType'] for stage, reqs in list(args.sys_reqs_from_clone.items())},
-                                  **(args.instance_type or {}))
+    requested_instance_type, requested_cluster_spec = {}, {}
+    executable_describe = None
 
-    if args.sys_reqs_from_clone and not isinstance(args.instance_count, str):
-        # extract instance counts from cloned sys reqs and override them with args provided with "dx run"
-        args.instance_count = dict({fn: reqs['clusterSpec']['initialInstanceCount']
-                                        for fn, reqs in list(args.sys_reqs_from_clone.items()) if 'clusterSpec' in reqs},
-                                   **(args.instance_count or {}))
+    from ..utils import merge
```

Comment:
I would consider placing this import on the top...or is there any specific reason to have it here? can this import break something (functionality, different OS, ...)?

Reply:
good point! I just feel like it is only used here so better to not import it always. But changed it as you suggested. Thanks!